undefined reference to `swift_isUniquelyReferenced_native'

### Description

A class containing an optional variable of the same class as itself causes a linker failure in embedded Swift `undefined reference to swift_isUniquelyReferenced_native`.

https://godbolt.org/z/cWd6h53Go

### Reproduction

```swift
class Foo {
    var foo: Foo?
}
```


### Expected behavior

No linker failure :-)

### Environment

Nightly

Apple Swift version 6.0-dev (LLVM 1ad249b55ef5cc0, Swift b10d97dd01fac82)
Target: arm64-apple-macosx14.0

### Additional information

@rauhul @kubamracek

--- a/stdlib/public/core/EmbeddedRuntime.swift
+++ b/stdlib/public/core/EmbeddedRuntime.swift
@@ -146,6 +146,12 @@ func swift_initStackObject(metadata: UnsafeMutablePointer<ClassMetadata>, object
 public func swift_setDeallocating(object: Builtin.RawPointer) {
 }
 
+@_cdecl("swift_isUniquelyReferenced_native")
+public func swift_isUniquelyReferenced_native(object: Builtin.RawPointer) -> Bool {
+  if Int(Builtin.ptrtoint_Word(object)) == 0 { return false }
+  return swift_isUniquelyReferenced_nonNull_native(object: UnsafeMutablePointer<HeapObject>(object))
+}
+
 @_cdecl("swift_isUniquelyReferenced_nonNull_native")
 public func swift_isUniquelyReferenced_nonNull_native(object: Builtin.RawPointer) -> Bool {
   return swift_isUniquelyReferenced_nonNull_native(object: UnsafeMutablePointer<HeapObject>(object))

--- a/test/embedded/classes-optional.swift
+++ b/test/embedded/classes-optional.swift
@@ -1,0 +1,19 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -enable-experimental-feature Embedded -c -o %t/main.o
+// RUN: %target-clang %t/main.o -o %t/a.out -dead_strip
+// RUN: %target-run %t/a.out | %FileCheck %s
+
+// REQUIRES: swift_in_compiler
+// REQUIRES: executable_test
+// REQUIRES: optimized_stdlib
+// REQUIRES: OS=macosx || OS=linux-gnu
+
+class Foo {
+    var foo: Foo?
+}
+
+do {
+  _ = Foo()
+  print("OK!")
+}
+// CHECK: OK!

